src/readinglistbar.cpp

```diff
@@ -13,8 +13,23 @@ ReadingListBar::ReadingListBar(QWidget *parent) :
     ui->setupUi(this);
     connect(KiwixApp::instance()->getLibrary(), &Library::bookmarksChanged,
             this, &ReadingListBar::setupList);
-    connect(ui->listWidget, &QListWidget::itemActivated,
-            this, &ReadingListBar::on_itemActivated);
+    connect(ui->listWidget, &QListWidget::itemClicked,
+            this, [this](QListWidgetItem* item) {
+                onItemClicked(item, QGuiApplication::mouseButtons());
+            });
+    connect(ui->listWidget, &QListWidget::itemDoubleClicked,
+            this, [this](QListWidgetItem* item) {
+                onItemDoubleClicked(item, QGuiApplication::mouseButtons());
+            });
+    connect(ui->listWidget, &QListWidget::itemPressed, 
+            this, [this](QListWidgetItem* item) {
+                onItemPressed(item, QGuiApplication::mouseButtons());
+            });
+    connect(ui->listWidget, &QListWidget::itemActivated, 
+            this, [this](QListWidgetItem* item) {
+                onItemActivated(item, QGuiApplication::mouseButtons());
+            });
```

Comment:
When you remove the `MouseButtons` parameter for some of the `onItem*()` handlers, please simplify the `connect()` expression.